models/ios/logging_global/ios_logging_global.yaml

```
@@ -0,0 +1,1071 @@
+              filtered:
+                description: Enable filtered logging
+                type: dict
+                suboptions:
+                  sequence_num_session: 
+                    description: Include session sequence number tag in syslog message
+                    type: int
+                  session_id: 
+                    description: Specify syslog message session ID tagging
+                    type: int
+                  stream: 
+                    description: This server should only receive messages from a numbered stream
+                    type: str
+              sequence_num_session:
+                description: Include session sequence number tag in syslog message
+                type: dict
+                suboptions:
+                  discriminator: 
+                    description:  Message discriminator identifier for this logging session
+                    type: str
+              session_id:
+                description: Specify syslog message session ID tagging
+                type: dict
+                suboptions:
+                  hostname: 
+                    description: Include hostname in session ID tag
+                    type: dict
+                    suboptions:
+                      discriminator: 
+                        description:  Message discriminator identifier for this logging session
+                        type: str
+                      sequence_num_session: 
+                        description:  Message discriminator identifier for this logging session
+                        type: str
+                  ipv4: 
+                    description: Include logging source IP address in session ID tag
+                    type: bool
+                  ipv6: 
+                    description: Include logging source IPV6 address in session ID tag
+                    type: bool
+                  string: 
+                    description: Include custom string in session ID tag
+                    type: str
+              transport:
+                description: Specify the transport protocol (default=UDP)
+                type: dict
+                suboptions:
+                  tcp: 
+                    description: Transport Control Protocol
+                    type: dict
+                    suboptions:
+                      audit:
+                        description: Set this host for IOS firewall audit logging
+                        type: dict
+                        suboptions:
+                          discriminator: 
+                            description:  Specify a message discriminator identifier for this logging session
+                            type: str
+                          sequence_num_session: 
+                            description:  Include session sequence number tag in syslog message
+                            type: dict
+                            suboptions:
+                              discriminator: 
```

I see this `discriminator` param being the single key param for sequence_num_session, can't you make it part of sequence_num_session itslef, like:

---

```
an-ios-02(config)#logging host 10.0.2.15 transport tcp audit discriminator word?
  <cr>
```

```
an-ios-02(config)#logging host 10.0.2.15  transport tcp audit sequence-num-session discriminator Word ?
  <cr>
```
@justjais  Both being valid commands, the is no other option to detect what is the parent command of discriminator

---

@KB-perByte saw the config and it's:
```
an-ios-01(config)#logging host 10.0.2.15 transport tcp audit sequence-num-session ?
  discriminator  Specify a message discriminator identifier for this logging session
  <cr>
```
which means either u can just set `sequence-num-session` or give `discriminator` value to the `sequence-num-session`, in that case your model should be like:
```suggestion
                         sequence_num_session: 
                            description: Include session sequence number tag in syslog message
                            type: dict
                            suboptions:
                              set:
                                description: set sequence-num-session
                                type: bool
                             discriminator: 
                                description: set sequence-num-session
                                type: str
```

---

@justjais 
```
an-ios-02(config)#logging host 10.0.2.15 transport tcp audit ?                     
  discriminator         Specify a message discriminator identifier for this logging session
  filtered              Enable filtered logging
  sequence-num-session  Include session sequence number tag in syslog message
  session-id            Specify syslog message session ID tagging
  xml                   Enable logging in XML
  <cr>

```
still not sure why we are combining discriminator and sequence-num-serrsion, they should be different.. I tried figuring  out a valid command I could not `Specified MD by the name ATTFLTR1 is not found.`

---

hmm @KB-perByte this is incorrect understanding for the cmd: `logging host 10.0.2.15 transport tcp` all the param available represent single param, so respective model should look like:
```
          transport:
            description: Specify the transport protocol (default=UDP)
            type: dict
            suboptions:
              tcp: 
                description: Transport Control Protocol
                type: dict
                suboptions:
                  audit: bool
                  discriminator: str
                  filtered:
                    stream: int
                  port: int
                  sequence-num-session: bool
                  session-id:
                    hostname, ipv4, ipv6, string
                  xml: bool
```